pkg/auth/service.go

```diff
@@ -698,6 +704,12 @@ func (s *DBAuthService) CreateCredentials(ctx context.Context, username string)
 }
 
 func (s *DBAuthService) AddCredentials(ctx context.Context, username, accessKeyID, secretAccessKey string) (*model.Credential, error) {
+	if !reValidAccessKeyID.MatchString(accessKeyID) {
```

Comment:
I don't understand what this check is intended to achieve.
1. IT does _not_ guarantee a valid AWS - style access ID,  those all start with A and then have some 3 special uppercase letters.
2. It does _not_ guarantee a high entropy secret key.
3. It prevents me ftom using some perfectly useful characters. 
Given that this is intended for fast environment setup, I'd understand very limited permissions on it. But this validation doesn't prevent future failures (everything works with any access key), so I don't see its upside.

Reply:
We generate the key/secret. The one place we access these valid is using the non-documented command-line argument to 'setup' where we use predefined key/secret and set an admin user.
The use case was a testing environment. Recently also Our everything bagel docker-compose where we use the example key/secret.
Recently one of the users updated the key to something simple (lakefs) which broke the s3 gateway - every request stopped working and the sig was failed to verify.
https://github.com/treeverse/lakeFS/blob/69828df0fe7c4ec0fa2deaa1f2d53025951137cb/pkg/gateway/sig/v4.go#L37
The header verification match a valid key to the request - Used the same verification at the lowest level we accept the key.
Did the same for the secret, @guy-har requested - used the information found here: https://aws.amazon.com/blogs/security/a-safer-way-to-distribute-aws-credentials-to-ec2

Note this code was added as a safe-guard from people setting a non-working key/secret and not a new way to enable key/secret values into the system.